specs/checking_account_spec.rb

```diff
@@ -22,59 +23,149 @@
+
+      updated_balance.must_equal expected_balance
     end
 
     it "Outputs a warning if the account would go below -$10" do
-      # TODO: Your test code here!
+      start_balance = 100.0
+      account = Bank::CheckingAccount.new(1337, start_balance)
+
+      account.balance.must_equal start_balance, "Outputs a warning if the balance would go below -$10"
+
     end
 
     it "Doesn't modify the balance if the account would go below -$10" do
-      # TODO: Your test code here!
+      start_balance = 100.0
+      withdrawal_amount = 120.0
+      account = Bank::SavingsAccount.new(1337, start_balance)
+
+      account.withdraw(withdrawal_amount)
+      account.balance.must_equal start_balance
     end
 
     it "Requires a positive withdrawal amount" do
-      # TODO: Your test code here!
+      account = Bank::CheckingAccount.new(1337, 100.0)
+
+      proc {
+        account.withdraw_using_check(-10)
+      }.must_raise ArgumentError
     end
 
     it "Allows 3 free uses" do
-      # TODO: Your test code here!
+      start_balance = 100.0
+      withdrawal_amount = 10.0
+      account = Bank::CheckingAccount.new(1337, start_balance)
+      new_balance = start_balance - (withdrawal_amount * 3)
+
+      3.times do
+        account.withdraw_using_check(withdrawal_amount)
+      end
+      account.balance.must_equal new_balance
     end
 
     it "Applies a $2 fee after the third use" do
-      # TODO: Your test code here!
+      start_balance = 100.0
+      withdrawal_amount = 10.0
+      account = Bank::CheckingAccount.new(1337, start_balance)
+      new_balance = start_balance - ((withdrawal_amount * 4) + 2)
+
+      4.times do
+        account.withdraw_using_check(withdrawal_amount)
+      end
+      account.balance.must_equal new_balance
     end
-  end
+end
 
   describe "#reset_checks" do
     it "Can be called without error" do
-      # TODO: Your test code here!
+      start_balance = 100.0
+      check_count = 0
+      account = Bank::CheckingAccount.new(1337, start_balance)
+      account.reset_checks.must_equal check_count
     end
 
     it "Makes the next three checks free if less than 3 checks had been used" do
-      # TODO: Your test code here!
+      start_balance = 100.0
+      withdrawal_amount = 10.0
+      account = Bank::CheckingAccount.new(1337, start_balance)
+      new_balance = start_balance - ((withdrawal_amount * 4))
+
+      1.times do
+        account.withdraw_using_check(withdrawal_amount)
+      end
+      account.reset_checks
+      3.times do
+        account.withdraw_using_check(withdrawal_amount)
+      end
+      account.balance.must_equal new_balance
     end
 
     it "Makes the next three checks free if more than 3 checks had been used" do
-      # TODO: Your test code here!
+      start_balance = 100.0
+      withdrawal_amount = 10.0
+      account = Bank::CheckingAccount.new(1337, start_balance)
+      new_balance = start_balance - ((withdrawal_amount * 7) + 4)
+
+      5.times do
```

Comment:
I really like the idea of using a `times` loop here to avoid writing the same code many times. Good work!